modules/tf_cloudbuild_workspace/cb.tf

```diff
@@ -52,17 +52,29 @@ locals {
   default_prefix = var.prefix != "" ? var.prefix : replace(var.tf_repo_dir != "" ? "${local.repo}-${var.tf_repo_dir}" : local.repo, "/", "-")
 
   # default substitutions
-  default_subst = {
+  default_subst = var.enable_worker_pool ? {
+    "_TF_SA_EMAIL"          = local.cloudbuild_sa_email,
+    "_STATE_BUCKET_NAME"    = local.state_bucket_name,
+    "_LOG_BUCKET_NAME"      = local.log_bucket_name,
+    "_ARTIFACT_BUCKET_NAME" = local.artifacts_bucket_name,
+    "_PRIVATE_POOL"         = var.worker_pool_id,
+    } : {
     "_TF_SA_EMAIL"          = local.cloudbuild_sa_email,
     "_STATE_BUCKET_NAME"    = local.state_bucket_name,
     "_LOG_BUCKET_NAME"      = local.log_bucket_name,
     "_ARTIFACT_BUCKET_NAME" = local.artifacts_bucket_name,
   }
```

Comment:
We can do a conditional merge to make this succinct `merge({....}, var.enable_worker_pool ? {"_PRIVATE_POOL"         = var.worker_pool_id} : {})`

Reply:
good catch :) 
Fixed.